analysers/analyser_osmosis_relation_cyclic.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+
+###########################################################################
+##                                                                       ##
+## Copyrights Osmose Project 2022                                        ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+from modules.OsmoseTranslation import T_
+from .Analyser_Osmosis import Analyser_Osmosis
+
+sql10 = """
+WITH RECURSIVE rcte AS (
+  -- initial 'seed' relation
+  SELECT DISTINCT ON (relation_id)
+    relation_id AS start_id,
+    member_id,
+    1 as nloops,
+    CONCAT('r', relation_id) AS path
+  FROM
+    relation_members
+  WHERE
+    member_type = 'R' AND
+    relation_locate(relation_id) IS NOT NULL -- can't display relations without way/node members
+
+  UNION ALL
+
+  -- loop children until no 'relation'-type members or relation back to start
+  SELECT
+    rcte.start_id,
+    child.member_id,
+    nloops + 1 AS nloops,
+    CONCAT(rcte.path, ' > r', child.relation_id) AS path
+  FROM
+    rcte
+    JOIN relation_members AS child ON
+      child.relation_id = rcte.member_id
+    WHERE
+      child.member_type = 'R' AND
+      child.relation_id != rcte.start_id AND
+      nloops < {maxloops} -- avoid getting stuck in strange constructions
+)
+SELECT DISTINCT ON (start_id)
+  start_id,
+  ST_AsText(relation_locate(start_id)),
+  path
+FROM
+  rcte
+WHERE
+  start_id = member_id AND
+  nloops < {maxloops}
```

Comment:
Always true, doesn't ?

Reply:
Indeed, no need to filter for nloops given the first condition